internal/services/controller/handlers/filters/onlypodoom/filter.go

```diff
@@ -0,0 +1,17 @@
+package onlypodoom
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"castai-agent/internal/castai"
+	"castai-agent/internal/services/controller/util"
+)
+
+func Filter(e castai.EventType, obj interface{}) bool {
+	k8sEvent, ok := obj.(*corev1.Event)
+	if !ok {
+		return false
+	}
+
+	return util.IsOOMEvent(k8sEvent)
```

Comment:
Here it looks like checking only for OOM event, but inside has more logic. Why is that logic moved to util, if we have a whole package here for that? Would be in single place and clear what it does...

Reply:
yep, the code was structured differently at the begging of the implementation, kind of a leftover, will remove.